getting_started_advanced.md

```diff
@@ -0,0 +1,276 @@
+```python
+class LedArray(GeneratorBlock):
+  @init_in_parent
+  def __init__(self, count: IntLike) -> None:
+    super().__init__()
+    self.ios = self.Port(Vector(DigitalSink.empty()), [Input])
+    self.gnd = self.Port(Ground.empty(), [Common])
+    self.generator(self.generate, count)
+    
+  def generate(self, count: int) -> None:
+    ...
+```
+
+> While here we use generators as a way to get a concrete value for circuit generation (the LED count), generators can also be used to do calculations beyond the operations available with the parameters.
+> For example, while we can add two `IntExpr`s (which produces another `IntExpr`), something more complex like square root is not provided.
+> For those cases, use a generator to get the parameter's value, where you have access to the full power of Python.
+
+### Port Arrays - Internal
+So far, the port array is still empty, so we must define its elements.
+With the count available as an int, we can use the `for` loop structure from before:
+```python
+class LedArray(GeneratorBlock):
+  ...
+  def generate(self, count: int) -> None:
+    for i in range(count):
+      self.ios.append_elt(DigitalSink.empty())
+      ...
+```
+
+> Port array's `.append_elt(...)` takes in the same arguments as `self.Port(...)`.
+> 
+> To mark a port array as explicitly having no elements, use `.defined()`.
+
+### Circuit Generation
+Instantiate the LEDs and connect them to the IO pin and ground as needed.
+`.append_elt(...)` returns the newly created port within the array, which can be used in `self.connect(...)`.
+
+> <details>
+>   <summary>At this point, your HDL might look like...</summary>
+>
+>   ```python
+>   class LedArray(GeneratorBlock):
+>     @init_in_parent
+>     def __init__(self, count: IntLike) -> None:
+>       super().__init__()
+>       self.ios = self.Port(Vector(DigitalSink.empty()), [Input])
+>       self.gnd = self.Port(Ground.empty(), [Common])
+>       self.generator(self.generate, count)
+>   
+>     def generate(self, count: int) -> None:
+>         self.leds = ElementDict[IndicatorLed]()
+>         for i in range(count):
+>           io = self.ios.append_elt(DigitalSink.empty())
+>           self.leds[i] = self.Block(IndicatorLed())
+>           self.connect(io, self.leds[i].signal)
+>           self.connect(self.gnd, self.leds[i].gnd)
+>   ```
+> </details>
+
+
+## Putting it All Together
+
+Replace the `for` loop in your top-level design with the single parameterized `LedArray` instantiation, and connect it to the microcontroller:  
+```python
+class BlinkyExample(SimpleBoardTop):
+  def contents(self) -> None:
+    ...
+    with self.implicit_connect(
+            ...
+    ) as imp:
+      ...
+      self.led = imp.Block(LedArray(4))
+      self.connect(self.mcu.gpio.request_vector('led'), self.led.ios)
+```
+
+Or, since the new block has both implicit scope and chain tags:
```

Comment:
I didn't really understand this line. Maybe an additional explanation of implicit scope and chain tags can be added.

Reply:
Clarified by referencing the `Input` tag, I'm hoping that will jog people's memory enough.

In the longer term we should probably write a reference document for these concepts, and link them from here.